### PR TITLE
Convert image-actions to typescript

### PR DESCRIPTION
**Pull request by mikedijkstra**
This PR converts image-actions to typescript.

- [x] Add typescript, eslint, prettier and related config
- [x] Add build, watch, format and lint scripts
- [x] Update `src/*.js` files to `src/*.ts` files
- [x] Update .gitignore
- [x] Update Dockerfile to build ts (`npm run build`)
- [x] Update entrypoint to run from `dist` folder
- [x] Update README
- [x] Update mozjpeg
- [x] Export missing `JPEG_PROGRESSIVE` constant

### CODE REVIEW

**Comment by benschwarz** [comment]
Woot! Thanks @mikedijkstra. Let's look to get this merged _swiftly!_ Do you want me to pickup some/all of the remaining tasks?

**Comment by mikedijkstra** [comment]
@benschwarz Yeah, conversion is done so if you're happy to get it over the line and make sure it actually runs as expected and release a new version that'd be great 👍️

File: package.json [hunk 1]
```diff
@@ -6,17 +6,36 @@
...
+  "prettier": {
+    "semi": false,
+    "singleQuote": true
   },
   "jest": {},
   "dependencies": {
     "@octokit/rest": "^16.17.0",
+    "@types/js-yaml": "^3.12.5",
     "glob": "^7.1.3",
+    "globby": "^11.0.1",
```
Comment (benschwarz) [inline package.json:25]: What's `globby` for? @mikedijkstra

Comment (mikedijkstra) [reply package.json:25]: Can be removed. I explored using it for better type support, but not required.

**Comment by benschwarz** [comment]
Did you have specific README notes in mind @mikedijkstra, or more just thinking about explaining "How to dev"?

File: package.json (hunk 1 above)
Comment (benschwarz) [reply package.json]: Gotcha.

Comment (mikedijkstra) [reply package.json]: Now that I think about it, it was for better promise support and not needing to use `util.promisify`, but again, not required!

**Comment by mikedijkstra** [comment]
> Did you have specific README notes in mind @mikedijkstra, or more just thinking about explaining "How to dev"?

Just thinking "How to dev".

File: src/image-processing.ts [hunk 2]
```diff
@@ -29,7 +29,7 @@ const processImages = async (): Promise<ProcessedImagesResult> => {
     ignore: config.ignorePaths.map((p: string) =>
       path.resolve(REPO_DIRECTORY, p)
     ),
-    onlyFiles: true
+    nodir: true
```
Comment (mikedijkstra) [inline src/image-processing.ts:32]: 👍️ Good pick up, I was just going to check I reverted this too!

**Comment by benschwarz** [comment]
Anything you can spot @mikedijkstra?

File: README.md [hunk 3]
```diff
@@ -147,14 +145,14 @@ jobs:
 
 ## Compressing images on a schedule
 
-It is also possible to run image-actions on a reoccurring schedule. By using the `compressOnly` option,  in conjunction with [@peter-evans's](/peter-evans) [`create-pull-request`](https://github.com/peter-evans/create-pull-request) action, a new Pull Request will be raised if there are optimised images in a repository.  
+It is also possible to run image-actions on a reoccurring schedule. By using the `compressOnly` option, in conjunction with [@peter-evans's](/peter-evans) [`create-pull-request`](https://github.com/peter-evans/create-pull-request) action, a new Pull Request will be raised if there are optimised images in a repository.
```
Comment (mikedijkstra) [inline README.md:148]:
```suggestion
It is also possible to run image-actions on a recurring schedule. By using the `compressOnly` option, in conjunction with [@peter-evans's](/peter-evans) [`create-pull-request`](https://github.com/peter-evans/create-pull-request) action, a new Pull Request will be raised if there are optimised images in a repository.
```

> Something that is recurring happens over and over again, possibly at regular intervals. In contrast, something that is reoccurring is simply happening again but not always repeatedly.
[source](https://www.grammarly.com/blog/recur-reoccur/)

Comment (benschwarz) [reply README.md]: Are you saying I shouldn't invent words? 🤪

**Comment by mikedijkstra** [comment]
> Anything you can spot @mikedijkstra?

Looks good to me!

**Pull request merged**